indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1271,6 +1277,134 @@ public void remove(final Task task)
     }
   }
 
+  /**
+   * Verifies that every lock held by the replacing task is a REPLACE lock
+   * @param task replacing task
+   */
+  public void verifyLocksForReplaceTask(Task task)
+  {
+    for (TaskLockPosse lockPosse : findLockPossesForTask(task)) {
+      if (lockPosse.taskLock.getType() != TaskLockType.REPLACE) {
+        throw DruidException.defensive(
+            "All the locks must be of type REPLACE for segmentTransactionalReplace. Found lock of type[%s] for task[%s].",
```

Comment:
Thanks for pointing it out. I think the message needs to be changed appropriately in that case.
If not, why can't this method be a part of the validation to acquire the transactional lock as was being done previously?

Reply:
@kfaraz the validations have been moved to the actions.

Reply:
>  If not, why can't this method be a part of the validation to acquire the transactional lock as was being done previously?

The validation really doesn't belong in `TaskLockbox` because `TaskLockbox` itself doesn't impose the restriction on any task holding a certain type of locks.